### PR TITLE
[libzippp] Update to libzippp 5.1-1.8.0

### PR DESCRIPTION
**Pull request by ctabin**
Upgrade to the latest [version](https://github.com/ctabin/libzippp/releases/tag/libzippp-v5.1-1.8.0) of libzippp.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
> _This is a new experimental fast check for PR issues. Please let us know if this bot is helpful!_
> 
> Use the version scheme "version" instead of "version-string" in port "libzippp".
> 
> After committing all other changes, the version database must be updated
> You have modified or added at least one portfile where deprecated functions are used.
> 
> If you feel able to do so, please consider migrating them to the new functions: `vcpkg_install_cmake` -> `vcpkg_cmake_install` (from port `vcpkg-cmake`) `vcpkg_build_cmake` -> `vcpkg_cmake_build` (from port `vcpkg-cmake`) `vcpkg_configure_cmake` -> `vcpkg_cmake_configure` (Please remove the option `PREFER_NINJA`) (from port `vcpkg-cmake`) `vcpkg_fixup_cmake_targets` -> `vcpkg_cmake_config_fixup` (from port `vcpkg-cmake-config`)
> 
> In the ports that use the new function, you have to add the corresponding dependencies:
> 
> ```json
> {
>   "name": "vcpkg-cmake",
>   "host": true
> },
> {
>   "name": "vcpkg-cmake-config",
>   "host": true
> }
> ```
> 
> The following files are affected:
> 
> * `ports/libzippp/portfile.cmake`
> 
> You have modified or added at least one vcpkg.json where you should check the `license` field.

@ctabin ,Thanks for your pr, could you please modernize your portfile.cmake according to the above tips?

File: ports/libzippp/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "libzippp",
-  "version-string": "5.0-1.8.0",
+  "version-string": "5.1-1.8.0",
   "port-version": 1,
```
Comment (JonLiu1993) [inline ports/libzippp/vcpkg.json:4]:
```suggestion
```

Comment (JonLiu1993) [reply ports/libzippp/vcpkg.json:4]: We should remove "port-version" when we update port version

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libzippp/vcpkg.json (hunk 1 above)
Comment (ctabin) [reply ports/libzippp/vcpkg.json]: Hi @JonLiu1993, that's removed, however I don't get how this port-version is about :thinking:

File: ports/libzippp/portfile.cmake [hunk 2]
```diff
@@ -22,13 +22,13 @@ vcpkg_configure_cmake(
...
 
 if(VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_fixup_cmake_targets(CONFIG_PATH "cmake/libzippp")
+    vcpkg_cmake_config_fixup(CONFIG_PATH "cmake/libzippp")
 else()
-    vcpkg_fixup_cmake_targets(CONFIG_PATH "share/libzippp")
+    vcpkg_cmake_config_fixup(CONFIG_PATH "share/libzippp")
 endif()
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
```
Comment (JonLiu1993) [inline ports/libzippp/portfile.cmake:34]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libzippp/portfile.cmake [hunk 3]
```diff
@@ -12,7 +12,7 @@ vcpkg_check_features(
         FEATURES    
         encryption LIBZIPPP_ENABLE_ENCRYPTION)
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/libzippp/portfile.cmake:17]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

Comment (JonLiu1993) [inline ports/libzippp/portfile.cmake:17]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/libzippp/portfile.cmake (hunk 2 above)
Comment (JonLiu1993) [inline ports/libzippp/portfile.cmake:34]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Feature "encryption" tested successfully in the following triplet:
- x86-windows
- x64-widnows
- x64-windows-static

**Comment by ras0219-msft** [comment]
Thanks for the upgrade!

**Pull request merged**